add terminal intellisense docs (#8559)
microsoft/vscode-docs: docs/terminal/shell-integration.md

IntelliSense can be triggered manually `kb(workbench.action.terminal.requestCompletions)` or by typing, which can be disabled with `setting(terminal.integrated.suggest.quickSuggestions)`. Intellisense can also be triggered when certain characters are typed, such as `/`, which can be configured with `terminal.integrated.suggest.suggestOnTriggerCharacters`.

When the shell provides an inline completion, VS Code surfaces this as the first completion item in the list. You can further configure this behavior with the `setting(terminal.integrated.suggest.inlineSuggestion)` setting.

The `setting(terminal.integrated.suggest.showStatusBar)` setting controls if a status bar shows up at the bottom of the list. This status bar provides actions like **Learn More** (`kb(workbench.action.terminal.suggestLearnMore)`), **Insert** (`kb(workbench.action.terminal.acceptSelectedSuggestion)`), and **Configure** (`kb(workbench.action.terminal.configureSuggestSettings)`). When you use the IntelliSense feature for the first several times, the **Learn More** action is highlighted for extra discoverability.

> The suggest widget can show details. You can toggle the detail widget's visiblity with `kb(workbench.action.terminal.suggestToggleDetails)`. Screen reader users can focus the details widget with `kb(workbench.action.terminal.suggestToggleDetailsFocus)` to hear them read out.

Reviewer: ```suggestion
The suggest control can show extra details about the suggestion. You can toggle the visibility of these details with `kb(workbench.action.terminal.suggestToggleDetails)`. Screen reader users can focus the details control with `kb(workbench.action.terminal.suggestToggleDetailsFocus)` to hear them read out.
```

Reviewer: We don't use the technical term "widget" and use "control" instead.

To view the shell integration quality, hover the terminal tab. Optionally, select **Show Details** on the hover to view more detailed information.

> ## IntelliSense (Preview)

Reviewer 2: @ntrogh should we move this back to the bottom until it's no longer preview? This is the right place to put it when it's no longer preview imo.

Reviewer: Agree. Preview features should go lower down the article.

> ### Background

Reviewer 2: We could remove this header

Reviewer: ☝️ Try to avoid empty headers - in this case, the background info can go directly under IntelliSense.

> IntelliSense, powered by shell integration `setting(terminal.integrated.shellIntegration.enable)`, can be enabled with `setting(terminal.integrated.suggest.enabled)`.

Reviewer: ```suggestion
IntelliSense in the terminal enables you to receive suggestions for files, folders, command, and command arguments and options. IntelliSense is powered by shell integration (`setting(terminal.integrated.shellIntegration.enable)`) and you can enable IntelliSense with the `setting(terminal.integrated.suggest.enabled)`. ```

Reviewer: Added a short intro sentence about what IntelliSense is

> By default, on request of completions `kb(workbench.action.terminal.requestCompletions)` files, folders, commands, arguments, and options are provided.

Reviewer: ```suggestion
By default, you can invoke IntelliSense by using the `kb(workbench.action.terminal.requestCompletions)` keyboard shortcut.
```

![A user has typed git checkout in a terminal and requests completions. The completion list shows branch names like main and merogge/terminal-section](images/shell-integration/terminal-suggest.png)

> Commands are sourced from [Fig specs](https://github.com/withfig) and per-shell builtin functions (for `pwsh`, `bash`, `zsh`, and `fish`) and are validated against the `$PATH` to ensure they exist. On Windows, the particular set of executables can be configured with `setting(terminal.integrated.suggest.windowsExecutableExtensions)`.

Reviewer: ```suggestion
VS Code sources commands from [Fig specs](https://github.com/withfig) and validates per-shell built-in functions (for `pwsh`, `bash`, `zsh`, and `fish`) against the `$PATH` to ensure they exist. On Windows, you can configure the particular set of executables with the `setting(terminal.integrated.suggest.windowsExecutableExtensions)` setting.
```

> There can be multiple IntelliSense providers, which can be changed with `setting(terminal.integrated.suggest.providers)`.

Reviewer: ```suggestion
The terminal supports multiple IntelliSense providers. You can configure them with the `setting(terminal.integrated.suggest.providers)` setting.
```

### Keyboard navigation

> By default, `Tab` inserts the suggestion. Once navigation of the list has occurred, `Enter` will similarly insert the suggestion. This can be configured with `setting(terminal.integrated.suggest.selectionMode)`.

Reviewer: ```suggestion
By default, `kbstyle(Tab)` inserts the suggestion. Once navigation of the list has occurred, `kbstyle(Enter)` will similarly insert the suggestion. You can configure this behavior with the `setting(terminal.integrated.suggest.selectionMode)` setting.
```

Reviewer: Just edited for style but I don't really understand what "once navigation of the list has occurred" means. You might want to clarify.

To both insert and run the completion in the terminal on acceptance, configure `setting(terminal.integrated.suggest.runOnEnter)`.

> IntelliSense is triggered when typing, which can be disabled with `setting(terminal.integrated.suggest.quickSuggestions)`. Intellisense can also be triggered when certain characters are typed, such as `/`, which can be configured with `terminal.integrated.suggest.suggestOnTriggerCharacters`.

Reviewer: If I disable triggering by typing, am I them completely disabling it or do I need to manually trigger them? Not clear when reading this sentence. 
It can also be triggered when typing a character. Is this then the alternative to typing or an extra thing on top of manually triggering a completion?

> When the shell provides an inline completion, we surface this as the first completion item and that can be configured with `setting(terminal.integrated.suggest.inlineSuggestion)`.

Reviewer: ```suggestion
When the shell provides an inline completion, VS Code surfaces this as the first completion item in the list. You can further configure this behavior with the `setting(terminal.integrated.suggest.inlineSuggestion)` setting.
```

> `setting(terminal.integrated.suggest.showStatusBar)` controls if a status bar shows up at the bottom of the list. This contains helpful actions like `Learn More` `kb(workbench.action.terminal.suggestLearnMore)`, `Insert` `kb(workbench.action.terminal.acceptSelectedSuggestion)`, and `Configure` `kb(workbench.action.terminal.configureSuggestSettings)`. When the feature is used for the first several times, the `Learn More` action will be highlighted for extra discoverability.

Reviewer: ```suggestion
The `setting(terminal.integrated.suggest.showStatusBar)` setting controls if a status bar shows up at the bottom of the list. This status bar provides actions like **Learn More** (`kb(workbench.action.terminal.suggestLearnMore)`), **Insert** (`kb(workbench.action.terminal.acceptSelectedSuggestion)`), and **Configure** (`kb(workbench.action.terminal.configureSuggestSettings)`). When you use the IntelliSense feature for the first several times, the **Learn More** action is highlighted for extra discoverability.
```

> Toggle suggest detail focus with `kb(workbench.action.terminal.suggestToggleDetails)` and focus the details widget with `kb(workbench.action.terminal.suggestToggleDetailsFocus)`.

Reviewer: "toggle suggest detail focus": having difficulties to understand what this is. Can you explain what this does?
